qipr_approver/approver/tests/tests_models/tests_models.py

```diff
@@ -8,61 +8,140 @@
 from django.core.urlresolvers import reverse
 from django.db.models import fields, signals
 from django.test import TestCase, Client
-from approver.tests.tests_models.test_utils import check_fields
+from approver.tests.tests_models.test_utils import check_fields, is_foreign_key_to
 
 class OrganizationModel(TestCase):
-    def test_organization_model(self):
-        org = Organization()
-        for field in Organization._meta.fields:
-            if field.name in "title":
-                self.assertTrue(isinstance(field, django.db.models.fields.CharField), True)
-                self.assertTrue(field.max_length, 30)
+    def setUp(self):
+        signals.post_save.disconnect(model_push, Organization)
+        self.user = User(username = 'testUser')
+        self.user.save()
+
+    def test_should_save_when_given_user(self):
+        org = Organization(org_name='test')
+        org.save(self.user)
+        self.assertIn(org, Organization.objects.all())
+
+    def test_should_not_save_when_missing_user(self):
+        org = Organization(org_name='test')
+        self.assertRaises(TypeError, lambda : org.save())
+
+    def test_should_delete_when_given_user(self):
+        org = Organization(org_name='test')
+        org.save(self.user)
+        if org in Organization.objects.all():
+            org.delete(self.user)
+            self.assertNotIn(org, Organization.objects.all())
+        else:
+            self.fail("Organization failed to save")
+
+    def test_should_not_delete_when_missing_user(self):
+        org = Organization(org_name='test')
+        self.assertRaises(TypeError, lambda : org.save())
+
+    def test_should_have_proper_fields_when_created(self):
+        self.assertTrue(check_fields(Organization, "org_name", "Char", 400))
+
+    def test_should_print_text_when_asked_for_string(self):
+        org_name = "test org"
+        org = Organization(org_name = org_name)
+
+        self.assertEqual(str(org), org_name)
+
+class TrainingModel(TestCase):
+    def test_should_have_proper_fields_when_created(self):
+        self.assertTrue(check_fields(Training, "name", "Char", 200))
 
 class SpecialityModel(TestCase):
```

Comment:
For all these tag-like models we might want to test that they inherit the right things. Not 100% sure how to do that but since the provenance feature, tag parsing and registry pushing all depend on what these models are inheriting, I think that it is an important test.